website/sidebars.json

```diff
@@ -81,32 +81,66 @@
       "direct-manipulation"
     ],
     "The New Architecture": [
-      "new-architecture-intro",
+      "the-new-architecture/intro-2",
+      "the-new-architecture/why",
+      "the-new-architecture/use-app-template",
       {
         "type": "category",
-        "label": "Supporting in your Library",
+        "label": "Pillars",
         "collapsible": false,
         "collapsed": false,
         "items": [
-          "new-architecture-library-intro",
-          "new-architecture-library-android",
-          "new-architecture-library-ios"
+          "the-new-architecture/pillars",
+          "the-new-architecture/pillars-turbomodules",
+          "the-new-architecture/pillars-fabric-components"
         ]
       },
       {
         "type": "category",
-        "label": "Supporting in your App",
+        "label": "Backward Compatibility",
         "collapsible": false,
         "collapsed": false,
         "items": [
-          "new-architecture-app-intro",
-          "new-architecture-app-modules-android",
-          "new-architecture-app-modules-ios",
-          "new-architecture-app-renderer-android",
-          "new-architecture-app-renderer-ios"
+          "the-new-architecture/backward-compatibility",
+          "the-new-architecture/backward-compatibility-turbomodules",
+          "the-new-architecture/backward-compatibility-fabric-components",
+          "the-new-architecture/backward-compatibility-troubleshooting"
+        ]
+      },
+      {
+        "type": "category",
+        "label": "Migration",
+        "collapsible": false,
+        "collapsed": false,
+        "items": [
+          "new-architecture-intro",
+          {
+            "type": "category",
+            "label": "Supporting in your Library",
+            "collapsible": false,
+            "collapsed": false,
+            "items": [
+              "new-architecture-library-intro",
+              "new-architecture-library-android",
+              "new-architecture-library-ios"
+            ]
+          },
+          {
+            "type": "category",
+            "label": "Supporting in your App",
+            "collapsible": false,
+            "collapsed": false,
+            "items": [
+              "new-architecture-app-intro",
+              "new-architecture-app-modules-android",
+              "new-architecture-app-modules-ios",
+              "new-architecture-app-renderer-android",
+              "new-architecture-app-renderer-ios"
+            ]
+          },
+          "new-architecture-troubleshooting"
         ]
       },
-      "new-architecture-troubleshooting",
       "new-architecture-appendix"
```

Comment:
Appendix and Troubleshooting are on different level which I'm not sure is correct. They are definitely both related to the same Migration section. As I see you added a 
`-troubleshooting` section for the backward compat section, maybe we can move the appendix also inside the migration section?

Ideally we could even work on nuking the appendix section as it risks to become an information silo 🤔

Reply:
I actually moved to the outer level on purpose, because there are topics valid also for the other sections.
For example, the [Flow Types](https://reactnative.dev/docs/next/new-architecture-appendix#i-flow-type-to-native-type-mapping) are valid also for the creation of new TM/FC other than for the migration.

Reply:
Gotcha. Ideally that appendix page should just be moved to a "Using the codegen" or somewhere else. What's your take on that?

Reply:
That is a clever idea. I was thinking about where to put such section about codegen, but I can't find a satisfying place, mainly because codegen is shared across `pillars`, `backward compatibility`, and the `migration`. It probably makes sense to have it as a sort of appendix in the `pillars` (which is the first section encountered by the users when reading the guide top to bottom) and to refer to it in the other two sections. What do you think?